core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -68,6 +68,8 @@
   private static final Logger log = Logger.getLogger(ClientCallImpl.class.getName());
   private static final byte[] FULL_STREAM_DECOMPRESSION_ENCODINGS
       = "gzip".getBytes(Charset.forName("US-ASCII"));
+  // visible for testing
+  static final long EXTRA_WAIT_TIME_BEFORE_SEND_CANCEL_IN_NS = 1_000_000;
```

Comment:
use `@VisibleForTesting`, can you use `TimeUnit.MILLIS.toNano(1)`? also, this deserves some explanation (can use the comment in the test).